pkg/registry/apps/deployment/storage/storage.go

```diff
@@ -279,10 +280,11 @@ func (r *ScaleREST) Get(ctx context.Context, name string, options *metav1.GetOpt
 
 // Update alters scale subset of Deployment object.
 func (r *ScaleREST) Update(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *metav1.UpdateOptions) (runtime.Object, bool, error) {
+	parentFieldManager := fieldmanager.FromContext(ctx)
 	obj, _, err := r.store.Update(
 		ctx,
 		name,
-		&scaleUpdatedObjectInfo{name, objInfo},
+		&scaleUpdatedObjectInfo{name, objInfo, parentFieldManager, options.FieldManager},
```

Comment:
Are we guaranteed that `options` is always non-nil here?

Reply:
It should be quite safe, options is created here, just before the Update call. Do you prefer me to add extra checks?